### PR TITLE
Several swift-api-digester fixes

### PR DESCRIPTION
**Pull request by nkcsgexi**

### CODE REVIEW

**Comment by weissi** [comment]
@nkcsgexi awesome! This looks like it would fix https://bugs.swift.org/browse/SR-10539 too, right?

**Comment by nkcsgexi** [comment]
@weissi yeah, I believe the second commit should fix it :)

File: tools/swift-api-digester/swift-api-digester.cpp [hunk 1]
```diff
@@ -2246,6 +2246,12 @@ static int prepareForDump(const char *Main,
   InitInvok.setMainExecutablePath(fs::getMainExecutable(Main,
     reinterpret_cast<void *>(&anchorForGetMainExecutable)));
   InitInvok.setModuleName("swift_ide_test");
+#if __APPLE__
+  InitInvok.getLangOptions().EnableObjCInterop = true;
+#else
+  // Ensure the tool works on linux properly
+  InitInvok.getLangOptions().EnableObjCInterop = false;
+#endif
```
Comment (jrose-apple) [inline tools/swift-api-digester/swift-api-digester.cpp]: :-( This should really depend on the target triple, the way it does in the driver.

Comment (nkcsgexi) [reply tools/swift-api-digester/swift-api-digester.cpp]: Fixed, mind taking another look?

Comment (jrose-apple) [reply tools/swift-api-digester/swift-api-digester.cpp]: Thank you!

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci Please smoke test Linux platform

**Pull request merged**